### PR TITLE
Feat: PLP multiple page template

### PR DESCRIPTION
**Pull request by eduardoformiga**
Depends on: this code will change after:

- https://github.com/vtex/faststore/pull/2215
- https://github.com/vtex/vtex-cicd-platform/pull/572
- https://github.com/vtex/faststore/pull/2227

## What's the purpose of this pull request?

This PR aims to find the best PLP template from the CMS, based on the PLP slug.

`vendemo` account
Screenshot 2024-02-06 at 18 44 14

## How it works?

It iterates the slug until there are no slashes (/), prioritizing the following order:
 
1. A PLP template that matches the subcategory (e.g. slug = /department/category/subcategory).
2. A PLP template that matches the category (e.g. slug = /department/category).
3. A PLP template that matches the department (e.g. slug = /department).
4. If no matches are found, use the generic PLP template.

## How to test it?

1. localhost Initial setup:
You can change the tenant from CMS to the account `vendemo`. This store has all the PLPs templates created and published. Also, we changed the stored to `lojavivara`, for the sake of testing with a more complete catalog without harming the devs using `storeframework` nowadays.

Screenshot 2024-02-23 at 13 26 42

Screenshot 2024-02-23 at 13 26 02

2. Note that we have different templates created for PLPs with all scenarios mentioned above. So, use the following PLPs to test

|Matching Type|URL|
|-|-|
|Slug Subcategory - /vivara/joias/aneis |http://localhost:3000/vivara/joias/aneis |
|Slug  Category - /vivara/joias |http://localhost:3000/vivara/joias |
|Slug  Department - /vivara |http://localhost:3000/vivara |
|General - /life |http://localhost:3000/life |

### Starters Deploy Preview

This PR in starter uses the same configs mentioned. `vendemo` for CMS and `lojavivara` for API:
- https://github.com/vtex-sites/starter.store/pull/389

You can test using the same URLs:

|Matching Type|URL|
|-|-|
|Slug Subcategory - /vivara/joias/aneis |https://sfj-047d489--starter.preview.vtex.app/vivara/joias/aneis |
|Slug  Category - /vivara/joias |https://sfj-047d489--starter.preview.vtex.app/vivara/joias |
|Slug  Department - /vivara |https://sfj-047d489--starter.preview.vtex.app/vivara |
|General - /life |https://sfj-047d489--starter.preview.vtex.app/life |

## References

Doc PR
- https://github.com/vtex/faststore/pull/2236

### CODE REVIEW

File: packages/core/src/utils/utilities.ts [hunk 1]
```diff
@@ -11,3 +13,53 @@ export function textToKebabCase(text: string): string {
...
+  
+  // Remove duplicate slashes and white spaces throughout the string
+  normalizedSlug = normalizedSlug.replace(/\/+/g, '/').replace(/\s+/g, '')
+  
+  return '/' + normalizedSlug.toLowerCase()
+}
+
+function findPLPTemplateBySlug(pages: PLPContentType[], slug: string) {
+  return pages.find((page) => {
+    // general PLP template
```
Comment (hellofanny) [inline packages/core/src/utils/utilities.ts:29]:
```suggestion
    // generic PLP template
```

**Review by hellofanny — COMMENTED** [review]
LGTM, I'll just wait to test it better later~

File: packages/core/faststore.config.default.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ module.exports = {
 
   // Platform specific configs for API
   api: {
-    storeId: 'storeframework',
+    storeId: 'lojavivara',
```
Comment (eduardoformiga) [inline packages/core/faststore.config.default.js:17]: 🎗️ 
```suggestion
    storeId: 'storeframework',
```

File: packages/core/src/server/cms.ts [hunk 3]
```diff
@@ -19,38 +14,23 @@ export type Options =
...
-  (typeof x.releaseId === 'string' || typeof x.documentId === 'string')
+  (typeof x.releaseId === 'string' ||
+    typeof x.documentId === 'string' ||
+    typeof x.versionId === 'string')
 
-export const getPage = async <T extends ContentData>(options: Options) => {
-  const result = await (isLocator(options)
+export const clientCMS = new ClientCMS({
+  workspace: config.api.workspace,
+  tenant: 'vendemo',
```
Comment (eduardoformiga) [inline packages/core/src/server/cms.ts:23]: 🎗️ 
```suggestion
  tenant: config.api.storeId,
```

File: packages/core/src/pages/[...slug].tsx [hunk 4]
```diff
@@ -106,35 +101,18 @@ export const getStaticProps: GetStaticProps<
...
-        versionId: page.versionId,
-      })
-    }
-  } else {
-    pageData = await getPage<PLPContentType>({
-      ...(previewData?.contentType === 'plp' ? previewData : null),
-      contentType: 'plp',
-    })
-  }
+  console.log('🚀 ~ cmsPage:', cmsPage)
```
Comment (eduardoformiga) [inline packages/core/src/pages/[...slug].tsx:115]: 🎗️ 
```suggestion
```

**Review by hellofanny — APPROVED** [review]
I've tested and it's working as expected 🚀 
well done forms!

**Pull request merged**